app/scripts/controllers/console.js

```diff
@@ -232,5 +281,107 @@ angular.module('console', ['sql'])
       }
     };
   })
+  .directive('lazyLoadScroll', function ($window) {
+    return {
+      restrict: 'A',
+      link: function (scope, element, attrs) {
+        var e = element[0];
+        $($window).bind('scroll', function () {
+          if (this.pageYOffset + this.innerHeight > e.scrollHeight) {
+              scope.$apply(attrs.lazyLoadScroll);
+          }
+        });
+      }
+    };
+  })
+  .directive('formattedArray', function ($filter) {
+    return {
+      restrict: 'E',
+      scope: {
+        array: '=',
+        typesarray: '=',
+        expand: '='
+      },
+      templateUrl: 'views/formatted-array.html',
+      link: function($scope) {
+
+        $scope.isExpanded = $scope.expand;
+
+        $scope.getArrayLength = function getArrayLength() {
+          return $scope.array.length;
+        };
+
+        $scope.formatArrayindex = function formatArrayIndex(index) {
+          return index + 1;
+        };
+
+        $scope.getArrayBaseType = function getArrayBaseType() {
+          if ($scope.typesarray[1] == 1) {
+            return typeof($scope.array[0]);
+          } else {
+            return $filter('columnTypeClass')($scope.typesarray[1]);
+          }
+        };
+
+        $scope.getArrayType = function getArrayType() {
+          return $scope.typesarray[0] == 101 ? 'Set' : 'Array';
+        };
+
+        $scope.isObject = function isObject(index) {
```

Comment:
these functions are defined twice.. you can put them in a service and reuse them in the directive, which will also be easier to test